src/auxiliary.jl

```diff
@@ -0,0 +1,15 @@
+threads_enabled = false
```

Comment:
```suggestion
const threads_enabled = Ref(false)
```
I think global variables should be `const` for performance reasons (even if not super necessary in this particular case). You then need to change access to it from `threads_enabled` to `threads_enabled[]`